addons/website_sale_autocomplete/static/src/js/addressForm.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module */
+
+import publicWidget from 'web.public.widget';
+import { DropPrevious } from 'web.concurrency';
+import { debounce } from "@web/core/utils/timing";
+
+
+publicWidget.registry.AddressForm = publicWidget.Widget.extend({
+    selector: '.oe_cart .checkout_autoformat',
+    events: {
+        'input input[name="street"]': '_onChangeStreet',
+        'click .js_autocomplete_result': '_onClickAutocompleteResult'
+    },
+
+    init: function() {
+        this.streetAndNumberInput = document.querySelector('input[name="street"]');
+        this.cityInput = document.querySelector('input[name="city"]');
+        this.zipInput = document.querySelector('input[name="zip"]');
+        this.countrySelect = document.querySelector('select[name="country_id"]');
+        this.stateSelect = document.querySelector('select[name="state_id"]');
+        this.dp = new DropPrevious();
+
+        this._onChangeStreet = debounce(this._onChangeStreet, 200);
+        this._super.apply(this, arguments);
+    },
+
+    _onChangeStreet: function(ev) {
+        if(!this.streetAndNumberInput.dataset.autocompleteEnabled) return;
```

Comment:
Could it be possible to create a selector that only loads if `autocompleteEnabled` is falsy? (untested)

```suggestion
    selector: '.oe_cart .checkout_autoformat input[name="street"][data-autocomplete-enabled="1"]',
    events: {
        'input input[name="street"]': '_onChangeStreet',
        'click .js_autocomplete_result': '_onClickAutocompleteResult'
    },

    init: function() {
        this.streetAndNumberInput = document.querySelector('input[name="street"]');
        this.cityInput = document.querySelector('input[name="city"]');
        this.zipInput = document.querySelector('input[name="zip"]');
        this.countrySelect = document.querySelector('select[name="country_id"]');
        this.stateSelect = document.querySelector('select[name="state_id"]');
        this.dp = new DropPrevious();

        this._onChangeStreet = debounce(this._onChangeStreet, 200);
        this._super.apply(this, arguments);
    },

    _onChangeStreet: function(ev) {
```